arangod/RestHandler/RestIndexHandler.cpp

```diff
@@ -67,9 +67,10 @@ LogicalCollection* RestIndexHandler::collection(
       } catch (...) {
       }
     } else {
-      return _vocbase->lookupCollection(cName);
+      return _vocbase->lookupCollection(cName).get();
```

Comment:
Why are you not converting this stuff to std::shared_ptr as well ?

Reply:
I tried to keep the changes to a minimum otherwise there are too many files in the changeset
although I do agree that eventually raw pointers should be shared pointers